### PR TITLE
feat(topic_manager_list) : add topic manager list query

### LINKED ISSUE #14: [FEATURE] Add list lookup service providers query handler support
## Desired Solution
The handler should orchestrate the incoming request flow, including the verification of the request body or query parameters, depending on the HTTP handler type. Please keep in mind that the contract definition examples are just placeholders for the real ones, which should be defined based on the `overlay-engine` API.  After the GorillaPool team provides the full contract of the `overlay-engine`, please make appropriate updated in the code. 

## Suggested Implementation

HTTP Method Type: **GET** 
Route:  _/api/v1/list-lookup-service-providers_
 
Example code snippet: 

```
package commands

import (
	"context"

	"github.com/gofiber/fiber/v2"
)

type LookupServiceMetadata struct {
	Name             string
	ShortDescription string
	IconURL          string
	Version          string
	InformationURL   string
}

type SupportedLookupServicesProvider interface {
	LookupServices(ctx context.Context) (map[string]LookupServiceMetadata, error)
}

type Handler struct {
	provider SupportedLookupServicesProvider
}

func (h *Handler) Handle(c *fiber.Ctx) error {
	......
}
```

[Code ref link - overlay-express list-topic-managers handler](https://github.com/bitcoin-sv/overlay-express/blob/8b4006dcce06d2b0e4a8818399d1754e7df0cd4d/src/OverlayExpress.ts#L606)

[Code ref link - the DTO to be returned](https://github.com/bitcoin-sv/overlay-services/blob/master/src/Engine.ts#L945)

## Testing 
Please ensure that all positive and negative scenarios are covered by unit tests.

### PR DESCRIPTION
**Pull request by cain4chain**
## Description of Changes

feat(topic_manager_list): add topic manager list query file

## Linked Issues / Tickets

https://github.com/4chain-ag/go-overlay-services/issues/13

## Testing Procedure

feat(topic_manager_list_test): add topic manager list query test file, empty, success, nil

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have run the linter

### CODE REVIEW

File: pkg/server/app/queries/topic_manager_list_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			InformationURL:   "https://example.com/info",
+		},
+		"manager2": {
+			ShortDescription: "Description 2",
+			IconURL:          "",
+			Version:          "",
+			InformationURL:   "",
+		},
+	}
+}
```
Comment (mgosek-4chain) [inline pkg/server/app/queries/topic_manager_list_test.go]: Let's extend the response a bit 

```
func (*TopicManagerListProviderAlwaysSuccess) ListTopicManagers() map[string]*queries.MetaData {
	return map[string]*queries.MetaData{
		"manager1": {
			ShortDescription: "Description 1",
			IconURL:          "https://example.com/icon.png",
			Version:          "1.0.0",
			InformationURL:   "https://example.com/info",
		},
		"manager2": {
			ShortDescription: "Description 2",
			IconURL:          "https://example.com/icon2.png",
			Version:          "1.0.0",
			InformationURL:   "https://example.com/info",
		},
	}
}
```

File: pkg/server/app/queries/topic_manager_list_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	assert.Equal(t, &version, result["manager1"].Version)
+	assert.Equal(t, &infoURL, result["manager1"].InformationURL)
+
+	// Check manager2 data
+	require.Contains(t, result, "manager2")
+	assert.Equal(t, "manager2", result["manager2"].Name)
+	assert.Equal(t, "Description 2", result["manager2"].ShortDescription)
+	assert.Nil(t, result["manager2"].IconURL)
+	assert.Nil(t, result["manager2"].Version)
+	assert.Nil(t, result["manager2"].InformationURL)
```
Comment (mgosek-4chain) [inline pkg/server/app/queries/topic_manager_list_test.go]: Let's add an extra helper function at the bottom of the file :) 

```
func toPtr[T any](x T) *T { return &x }
```

Then we can refactor the assertion section: 

```
	require.NoError(t, err)
	defer res.Body.Close()
	require.Equal(t, http.StatusOK, res.StatusCode)
	require.Equal(t, "application/json", res.Header.Get("Content-Type"))

	var actual queries.TopicManagerListHandlerResponse
	require.NoError(t, jsonutil.DecodeResponseBody(res, &actual))

	expected := queries.TopicManagerListHandlerResponse{
		"manager1": queries.TopicManagerMetadata{
			Name:             "manager1",
			ShortDescription: "Description 1",
			IconURL:          toPtr("https://example.com/icon.png"),
			Version:          toPtr("1.0.0"),
			InformationURL:   toPtr("https://example.com/info"),
		},
		"manager2": queries.TopicManagerMetadata{
			Name:             "manager2",
			ShortDescription: "Description 2",
			IconURL:          toPtr("https://example.com/icon2.png"),
			Version:          toPtr("1.0.0"),
			InformationURL:   toPtr("https://example.com/info"),
		},
	}

	assert.EqualValues(t, expected, actual)
```

File: pkg/server/app/queries/topic_manager_list_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+	// Then:
+	require.NoError(t, err)
+	defer res.Body.Close()
+	require.Equal(t, http.StatusOK, res.StatusCode)
+	require.Equal(t, "application/json", res.Header.Get("Content-Type"))
+
+	var result map[string]queries.TopicManagerMetadata
+	require.NoError(t, json.NewDecoder(res.Body).Decode(&result))
+	assert.Empty(t, result)
```
Comment (mgosek-4chain) [inline pkg/server/app/queries/topic_manager_list_test.go]:
```
        var actual queries.TopicManagerDocumentationHandlerResponse
	require.NoError(t, jsonutil.DecodeResponseBody(res, &actual))
	assert.Empty(t, actual)
```

**Review by mgosek-4chain — APPROVED** [review]
Minor: Added hints to improve readability. Overall, great work, Cain!

File: pkg/server/app/queries/topic_manager_list_test.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+func TestNewTopicManagerListHandler_WithNilProvider(t *testing.T) {
+	// Given:
+	var provider queries.TopicManagerListProvider = nil
+
+	// When & Then:
+	assert.Panics(t, func() {
+		queries.NewTopicManagerListHandler(provider)
+	}, "Expected panic when provider is nil")
+}
```
Comment (mgosek-4chain) [inline pkg/server/app/queries/topic_manager_list_test.go:107]: Let's add an extra helper function at the bottom of the file :) 

```
func toPtr[T any](x T) *T { return &x }
```

File: pkg/server/app/queries/topic_manager_list_test.go (hunk 2 above)
Comment (cain4chain) [reply pkg/server/app/queries/topic_manager_list_test.go]: Thanks, much cleaner approach :)

File: pkg/server/app/queries/topic_manager_list_test.go (hunk 3 above)
Comment (cain4chain) [reply pkg/server/app/queries/topic_manager_list_test.go]: Ahh sorry missed the util function, will add that in.

**Review by mgosek-4chain — APPROVED** [review]
